.github/workflows/github-metadata.yml

```diff
@@ -0,0 +1,23 @@
+name: github_metadata
+on:
+  pull_request_target:
+    types: [opened, reopened, labeled, unlabeled, synchronize]
+
+jobs:
+  check:
+    name: protect github metadata
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'github-metadata-change') }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    env:
+      GIT_DIR: git-dir.git
+    steps:
+      - name: Fetch PR
+        run: |
+          set -x
+          git init -b main
+          git remote add origin "${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}"
+          git fetch origin '${{ github.event.pull_request.head.sha }}'
```

Comment:
Hmmm.. You have a really strict timeout. Could this perhaps just do something like

    curl --silent --show-error -L https://github.com/cockpit-project/cockpit/pull/15771.diff | grep '^+++ .*/\.github/'

(like for [this PR](https://github.com/cockpit-project/cockpit/pull/15771.diff)) instead of cloning the whole repo just for this?

Reply:
That seems a bit racy....  I'd prefer to run on the object we've been handed to run on....

Reply:
Can you then please at least do shallow fetches of the two things you want to compare? And bump the timeout a bit?

Reply:
Repeating in new review, as this is labeled as outdated.